code/modules/codex/entries/ammunition.dm

```diff
@@ -0,0 +1,51 @@
+/datum/codex_category/ammo/New()
+
+	for(var/thing in subtypesof(/datum/ammo))
+		var/datum/ammo/ammo = new thing()
+		if(initial(ammo.hidden_from_codex))
```

Comment:
dont instance it if its hidden from codex, use the typepath with initial to check this THEN use new